### PR TITLE
Update middleware

### PR DESCRIPTION
**Pull request by kareemk**

### CODE REVIEW

File: lib/promiscuous/middleware.rb [hunk 1]
```diff
@@ -8,33 +8,21 @@ module Controller
...
-            Thread.current[:promiscuous_controller] = nil
-          end
-        end
-      end
+      Promiscuous::Middleware.with_context(full_name, options) { super }
     end
 
     def render(*args)
-      Promiscuous::Middleware.without_context { super }
+      Promiscuous::Middleware.with_context(full_name, options) { super }
```
Comment (nviennot) [inline lib/promiscuous/middleware.rb]: dont' change that. it was good before.

Comment (kareemk) [inline lib/promiscuous/middleware.rb]: why? isn't that assuming that writes will never happen in a view?

Comment (nviennot) [inline lib/promiscuous/middleware.rb]: yes, which is a good assumption.

Comment (kareemk) [inline lib/promiscuous/middleware.rb]: what about something along the lines of tracking last_visited_at?

**Pull request closed without merging**